src/actions/wapeSale/purchaseDomains.ts

```diff
@@ -106,7 +92,24 @@ export const purchaseDomains = async (
       );
   }
 
-    const tx = await contract
+  let tx: ethers.ContractTransaction;
+  const purchased = await contract.domainsPurchasedByAccount(address);
+
+  if (status === SaleStatus.PrivateSale) {
+    let userClaim: Maybe<Claim> = mintlist.claims[address];
+
+    // To purchase in private sale a user must be on the mintlist
+    errorCheck(userClaim === undefined, "User is not part of private sale");
+    userClaim = userClaim!;
+  
+    // Cannot purchase over the allowed mintlist limit
+    errorCheck(
+      purchased.add(count).gt(userClaim.quantity),
+      `This user has already purchased ${purchased.toString()} and buying ${count.toString()} more domains would go over the
+      maximum purchase amount of domains for this user, ${userClaim.quantity}. Try reducing the purchase amount.`
+    );
+
+    tx = await contract
```

Comment:
In my view, returning directly would be much clear rather than assigning to `tx` variable and returninig.

Reply:
I usually prefer the opposite side of your argument. It is easier when debugging something to be able to see the output of a function by assigning it to a variable. Doing `return func()` means you're unable to do this.  

If this were Solidity I'd agree, as it is beneficial to reduce memory costs by not assigning values to variables unless you have to, but I'd rather leave it here.